packages/remix-dev/config.ts

```diff
@@ -329,6 +338,15 @@ export interface RemixConfig {
    */
   serverPlatform: ServerPlatform;
 
+  /**
+   * Configuration of server bundles to produce. If this is defined then the
+   * top-level `serverBuildPath` value is ignored.
+   */
+  serverBundles?: {
+    serverBuildPath: string;
+    routes: RouteManifest;
```

Comment:
Routes should for sure accept some kind of pattern IMO, maybe a blob? Alternatively, for maximum flexibility, we could accept a function here. eg.

```js
serverBundles: (path) => path.startsWith("app/") ? "build/logged-in.js" : "build/logged-out.js",
```

That would also mean Remix wouldn't have to implement the matching internally, just leave it up to the user to return a build path for each route path.

Reply:
That's an interesting idea. I think the function would be defined on `routes` instead though, and return a boolean. I imagine there might end up being more properties on the bundle object that would affect the output (`server`, `serverBuildTarget`, etc) which would make your syntax not quite work.

Reply:
That could be done by returning an object from the function.